### PR TITLE
Generalize sizes and strides policy on _make_wrapper_subclass

### PR DESCRIPTION
**Pull request by suo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #78646

Previously, there was a `dispatch_strides` boolean arg. Change this to
a string argument that directly maps onto `SizesStridesPolicy`.

### CODE REVIEW

File: torch/csrc/autograd/python_variable.cpp [hunk 1]
```diff
@@ -240,6 +240,22 @@ class PyInterpreterHolder {
...
+
+  if (arg == "sizes") {
+    return c10::TensorImpl::SizesStridesPolicy::CustomSizes;
+  }
+
+  TORCH_CHECK_VALUE(
+      false,
+      "Unknown sizes_strides_policy: ",
+      arg,
+      "; expected 'strides' or 'sizes'");
```
Comment (albanD) [inline torch/csrc/autograd/python_variable.cpp:257]: Is there some doc about what each values does?

**Review by albanD — COMMENTED** [review]
Should we have at least one test for the size case?

File: torch/csrc/autograd/python_variable.cpp (hunk 1 above)
Comment (suo) [reply torch/csrc/autograd/python_variable.cpp:257]: I can add documentation for it, where is are the docs for `_make_wrapper_subclass`/`_make_subclass` today?

As for the mismatch—I don't think that's fixable within the scope of this PR (and it already exists). But hopefully we'll finish coverage of the various tensorimpl methods soon and so at least we can document it?

**Review by ezyang — APPROVED** [review]

File: torch/csrc/autograd/python_variable.cpp (hunk 1 above)
Comment (suo) [reply torch/csrc/autograd/python_variable.cpp:257]: hmmmm, I feel like it's weird to put the rules for what this argument does in the error message; we should probably just have like a real docstring? 

Okay I'm going to merge this now, but I will take as a follow up:
- Finding the right place to describe the rules
- Making the code actually reflect the rules (e.g. make more things overridable from torchdispatch)

**Pull request closed without merging**